### ISSUE TITLE
Invalid LockedTransfer message - BUG

### ISSUE THREAD

**Issue by offerm** [raiden-network/raiden#4039]
@rakanalh 
My setup has two raiden nodes A and B and a channel between them.

I'm sending from A to B:

first a transfer where A does not have the secret and B is not working with a resolver. This results in a `LockedTransfer` message from A to B. B respond with `SecretRequest` which is ignored by A. 
```
curl  -X POST http://localhost:$Aport/api/v1/payments/$token/$B -H 'Content-Type: application/json' --data-raw '{"amount": 2, "secret_hash": "0xa2abfe43b070b0f93b69441cec26290f20fda5c5b08e3a0a02d4674df348a270"}'
```

Next A is making a "regular" payment which ends ok.

```
curl  -X POST http://localhost:$Aport/api/v1/payments/$token/$B -H 'Content-Type: application/json' --data-raw '{"amount": 1}'
```

Next I **restart** A 

After the restart I try again a "regular" payment. This payment does not work.

```
curl  -X POST http://localhost:$Aport/api/v1/payments/$token/$B -H 'Content-Type: application/json' --data-raw '{"amount": 3}'
```
B's log shows this:
```
{
  "node": "133499f5",
  "raiden_events": [
    "{\"payment_identifier\": \"16919130081996907240\", \"reason\": \"Invalid LockedTransfer message. Balance proof's locksroot didn't match, expected: 0x9c8400c552dd8ff3ef2e84a81e6c4e6997f1203d82977827e7ebcc034e4543ca got: 0xbe13150d7acd878707ea5a2b0dac5ad341bd59473a29865e040cb23e576a9cca.\", \"_type\": \"raiden.transfer.events.EventInvalidReceivedLockedTransfer\", \"_version\": 0}",
    "{\"identifier\": \"16919130081996907240\", \"secrethash\": \"0x54386bef1ad37bc8919ce5c2d2b2111d85104e9589b77d8ceca4ddd7dedd5bef\", \"reason\": \"Invalid LockedTransfer message. Balance proof's locksroot didn't match, expected: 0x9c8400c552dd8ff3ef2e84a81e6c4e6997f1203d82977827e7ebcc034e4543ca got: 0xbe13150d7acd878707ea5a2b0dac5ad341bd59473a29865e040cb23e576a9cca.\", \"_type\": \"raiden.transfer.mediated_transfer.events.EventUnlockClaimFailed\", \"_version\": 0}"
  ],
  "event": "Raiden events",
  "logger": "raiden.raiden_service",
  "level": "debug",
  "greenlet_name": "GMatrixClient._sync user_id:@0x133499f529a8d7cf00d8c5f7e73841ddb62b9ca3:raidentransport.ki-decentralized.com sync_token:s6022409_1452461_0_1_2084_1_1_26708_1",
  "timestamp": "2019-05-09 14:13:15.299488"
}
```

Looks like when the revents replaying at A restart something is not working as it was before. 
Let me know if you can spot it.

Thanks

Attached are 3 logs. 2 from A and 1 from B.
[raiden-debug_2019-05-09T14:09:50.118723.log](https://github.com/raiden-network/raiden/files/3162721/raiden-debug_2019-05-09T14.09.50.118723.log)
[raiden-debug_2019-05-09T14:02:17.182977.log](https://github.com/raiden-network/raiden/files/3162723/raiden-debug_2019-05-09T14.02.17.182977.log)

[raiden-debug_2019-05-09T14:02:53.524486.log](https://github.com/raiden-network/raiden/files/3162726/raiden-debug_2019-05-09T14.02.53.524486.log)

**Comment by offerm** [comment]
Update: I found the reason. Working on a solution.

**Comment by offerm** [comment]
The reason for the problem is that secrethash was not saved/restored properly for TransferDescriptionWithSecretState.

@rakanalh @hackaugusto please let me know if you are OK with such a fix or you like to see a migration for this (given that we have this problem in master). 

Let me know if I should open a PR for this commit on master.

**Comment by rakanalh** [comment]
@offerm Thanks... we've been trying to get to this over different periods and didn't look into it in details until now.

IMO, this problem no longer exists because the code that was producing the problem was removed when we changed our states / state changes & events to dataclasses. Right now, all fields that exist in the dataclass are serialized and therefore this issue won't pop up provided that we serialize / deserialize **all** fields in any given dataclass.
